share/shwap/row.go

```diff
@@ -35,14 +34,12 @@ func NewRow(halfShares []share.Share, side RowSide) Row {
 
 // RowFromEDS constructs a new Row from an Extended Data Square based on the specified index and
 // side.
-func RowFromEDS(square *rsmt2d.ExtendedDataSquare, idx int, side RowSide) Row {
-	sqrLn := int(square.Width())
-	shares := square.Row(uint(idx))
+func RowFromShares(shares []share.Share, side RowSide) Row {
 	var halfShares []share.Share
 	if side == Right {
-		halfShares = shares[sqrLn/2:] // Take the right half of the shares.
+		halfShares = shares[len(shares)/2:] // Take the right half of the shares.
```

Comment:
I assume we can be confident these divides will never blow up?

Reply:
It is safe even for 0 len

Reply:
Len of shares is guaranteed to be power of 2, so should not be an issue, however we should be very cautious with that